### PR TITLE
Use UserId instead of AnonymousId while sending telemetry data

### LINKED ISSUE #4571: Telemetry should use userId instead of anonymousId
/kind bug

## How did you run odo exactly?
I haven't run anything

## Actual behavior
Telemetry events sent to segment currently use the `anonymousId` event property. 

## Expected behavior
Telemetry events should actually use `userId` (see https://github.com/redhat-developer/vscode-redhat-telemetry/blob/main/src/services/reporter.ts#L28) instead, so they can be properly tracked in Woopra.
This is the exact same data, but sent using a different property

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this?**
/kind code-refactoring

**What does this PR do / why we need it**:
This PR changes AnonymousId to UserID so that odo data can be synchronized with data from other projects.

**Which issue(s) this PR fixes**:

Fixes #4571 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
Send some data to Segment and check if `UserId` appears in the response.

### CODE REVIEW

File: pkg/segment/segment.go [hunk 1]
```diff
@@ -79,8 +79,8 @@ func (c *Client) Upload(action string, duration time.Duration, err error) error
 
 	// queue the data that helps identify the user on segment
 	if err1 := c.SegmentClient.Enqueue(analytics.Identify{
-		AnonymousId: anonymousID,
-		Traits:      addConfigTraits(),
+		UserId: anonymousID,
```
Comment (dharmit) [inline pkg/segment/segment.go:82]:
```suggestion
		UserId: userID,
```

File: pkg/segment/segment.go [hunk 2]
```diff
@@ -99,9 +99,9 @@ func (c *Client) Upload(action string, duration time.Duration, err error) error
 
 	// queue the data that has telemetry information
 	return c.SegmentClient.Enqueue(analytics.Track{
-		AnonymousId: anonymousID,
-		Event:       action,
-		Properties:  properties,
+		UserId:     anonymousID,
```
Comment (dharmit) [inline pkg/segment/segment.go:102]:
```suggestion
		UserId:     userID,
```

**Review by dharmit — CHANGES_REQUESTED** [review]
Might seem like a nitpick right now but, renaming it would help avoid confusion when someone looks at the code later.

**Comment by valaparthvi** [comment]
>Might seem like a nitpick right now but, renaming it would help avoid confusion when someone looks at the code later.

I can't say I disagree, but `anonymousId` as a variable name makes more sense in a way that it tells that the UserId has to be anonymous. I have half a mind of renaming it to `userId` though :sweat_smile:

**Comment by kadel** [comment]
> I can't say I disagree, but `anonymousId` as a variable name makes more sense in a way that it tells that the UserId has to be anonymous. I have half a mind of renaming it to `userId` though 😅

but the function that retrieves it is called `getUserIdentity` ;

**Review by dharmit — APPROVED** [review]

**Pull request merged**